src/cache/cache.rs

```diff
@@ -327,26 +327,75 @@ pub trait Storage: Send + Sync {
     async fn max_size(&self) -> Result<Option<u64>>;
 }
 
+/// Implement storage for operator.
+#[cfg(any(feature = "s3", feature = "azure"))]
+#[async_trait]
+impl Storage for opendal::Operator {
+    async fn get(&self, key: &str) -> Result<Cache> {
+        match self.object(&normalize_key(key)).read().await {
+            Ok(res) => {
+                let hit = CacheRead::from(io::Cursor::new(res))?;
+                Ok(Cache::Hit(hit))
+            }
+            Err(e) => {
+                warn!("Got AWS error: {:?}", e);
```

Comment:
aws only ?

Reply:
opendal::Error already contains it's service type.